### PR TITLE
fix(webapp.#935): use libcurl to send POST data so that HTTPS can be suppo…

### LINKED ISSUE #935: Https connection problem to web-money-manager-ex instance
make sure these boxed are checked before submitting your issue - thank you!
- Check OS
  - [x] Windows
  - [ ] Mac OSX
  - [ ] Linux 
-  Check MMEX version
  - [ ] 1.3.x
  - [x] 1.2.7
  - [ ] 1.1 or older

Hi there

I just installed https://github.com/moneymanagerex/web-money-manager-ex on my hosting. Then I installed a new Let's encrypt ssl certificate and redirect all http requests to https with the following apache RewriteCondition in the .htaccess file:

```
# Force https
RewriteCond %{HTTPS} !=on
RewriteRule ^.*$ https://%{SERVER_NAME}%{REQUEST_URI} [R=301,L]
```

Back in MMEX i correct the server settings to the following:
image

After a restart, when I try to import the data from the Web-Tool, this error appears:
image

Seems MMEX ignores the https protocol and try to acces the Web-Tool via http and don't follow the redirect?

Maybe this have smth. to do with https://github.com/moneymanagerex/moneymanagerex/issues/893

I don't want my Data to be transferred unencrypted.

### PR DESCRIPTION
**Pull request by Rocketeer007**
Remove the code that silently replaces `https://` with `http://`
Use libcurl (which supports SSL natively) to send the POST data

This finally resolves #935, which was closed some time ago despite not being fully fixed.

---
This change is [Reviewable](https://reviewable.io/reviews/moneymanagerex/moneymanagerex/1755)

### CODE REVIEW

**Comment by slodki** [comment]
Could you use PROXYIP, PROXYPORT, NETWORKTIMEOUT, USERAGENT as in https://github.com/moneymanagerex/moneymanagerex/blob/master/src/util.cpp#L191 ?
Is this function generic? Maybe it's worth to move it to util.h to use in different MMEX parts and call it from mmWebApp::WebApp_SendJson method?

**Comment by slodki** [comment]
There is one more place without HTTPS support in util.h: download_file() to be rewritten using libcurl...

**Comment by GitCop** [comment]
Thanks for your pull request while there were the following issues

 * Commit: b9f08462a5ec7e97d39ca66d0f56c8e318f0d21d
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 88a2c96c96f5ae79e8d2717353db506e60da299a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: a8bf70f2efb2a22031a5550a11c4ceb22c166baa
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f5eb2a516a93e3eb39237c0117e81fab32137b15
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for your pull request while there were the following issues

 * Commit: b9f08462a5ec7e97d39ca66d0f56c8e318f0d21d
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 88a2c96c96f5ae79e8d2717353db506e60da299a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: a8bf70f2efb2a22031a5550a11c4ceb22c166baa
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: f5eb2a516a93e3eb39237c0117e81fab32137b15
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 07ddadcb5481a5bc075e7be45e23efe40d669234
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 96c5419143dce15a041be28c5deecd9867f28750
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by Rocketeer007** [comment]
I rather naively assumed that all PRs would be "squash merged", so my comments on the later commits wouldn't be important or have to meet with the commit naming conventions for this project...

What's the preferred approach here? Should I squash these myself on my fork and submit a brand new PR, or is there a better way?

File: src/util.h [hunk 1]
```diff
@@ -153,7 +153,7 @@ bool get_crypto_currency_prices(std::vector<wxString>& symbols, double& usd_rate
 const bool getNewsRSS(std::vector<WebsiteNews>& WebsiteNewsList);
 CURLcode http_get_data(const wxString& site, wxString& output);
 CURLcode http_post_data(const wxString& site, const wxString& data, const wxString& contentType, wxString& output);
-bool download_file(const wxString& site, const wxString& path);
+bool http_download_file(const wxString& site, const wxString& path);
```
Comment (slodki) [inline src/util.h]: are you sure it will not work with ftp:// or https:// protocols?

File: src/util.h [hunk 2]
```diff
@@ -151,7 +151,7 @@ bool get_crypto_currency_prices(std::vector<wxString>& symbols, double& usd_rate
     , wxString& output);
 
 const bool getNewsRSS(std::vector<WebsiteNews>& WebsiteNewsList);
-CURLcode site_content(const wxString& site, wxString& output);
+CURLcode http_get_data(const wxString& site, wxString& output);
```
Comment (slodki) [inline src/util.h]: this will work with ftp:// and https:// too IMHO

File: src/util.h (hunk 1 above)
Comment (slodki) [inline src/util.h]: I think we can return CURLcode here in place in meaningless bool - MMEX code can select different actions depending on problem type or notify user about selected problems.

**Comment by slodki** [comment]
I think you should squash all commits in this PR - if you preserve branch name GitHub will replace this PR content with rewritten git history.

There is "Squash and megre" option for accepting PR in GH GUI but I prefer to manually fine-tune using local git.

**Comment by Rocketeer007** [comment]
I think this is now complete; all the comments that were previously made have (I think) been addressed, and I've sqashed it to a single commit.

**Comment by slodki** [comment]
Thx

**Pull request merged**